tools/PI/DevHome.PI/Models/AppRuntimeInfo.cs

```diff
@@ -62,4 +40,47 @@ public partial class AppRuntimeInfo : ObservableObject
 
     [ObservableProperty]
     private Visibility visibility = Visibility.Visible;
+
+    public ObservableCollection<FrameworkType> FrameworkTypes { get; } = [];
+
+    public AppRuntimeInfo()
+    {
+        // Note these are in alphabetical order of Name.
+        FrameworkTypes.Add(new FrameworkType("Avalonia.Base.dll", "Avalonia"));
+        FrameworkTypes.Add(new FrameworkType("DXCore.dll", "DirectX"));
+        FrameworkTypes.Add(new FrameworkType("Microsoft.Maui.dll", "Maui"));
+        FrameworkTypes.Add(new FrameworkType("MFC", "MFC", false));
+        FrameworkTypes.Add(new FrameworkType("Python.exe", "Python"));
+        FrameworkTypes.Add(new FrameworkType("Microsoft.Windows.SDK.NET.dll", "Windows App SDK"));
+        FrameworkTypes.Add(new FrameworkType("System.Windows.Forms.dll", "Windows Forms"));
+        FrameworkTypes.Add(new FrameworkType("Microsoft.WinUI.dll", "WinUI"));
+        FrameworkTypes.Add(new FrameworkType("PresentationFramework.dll", "WPF"));
+    }
+
+    public void CheckFrameworkTypes(string moduleName)
+    {
+        foreach (var item in FrameworkTypes)
+        {
+            // Skip if already matched.
+            if (item.Status == true)
```

Comment:
With a new targetprocess, we always create a new AppRuntimeInfo object. In what scenario will we have already matched the frameworktype?

Reply:
We're doing the check for every module. For cases where the match pattern is a partial match, we might already have matched say mfc123.dll and now we're seeing say mfc456.dll.